<!DOCTYPE html>
<HTML>
	
<!-- Mirrored from www.homeandlearn.co.uk/csharp/csharp_s5p6.html by HTTrack Website Copier/3.x [XR&CO'2005], Thu, 24 Jan 2013 06:55:55 GMT -->
<HEAD>
		<TITLE>Visual C# .NET - Try ... Catch</TITLE>
	<LINK REL=Stylesheet TYPE ="text/css" HREF="two_col_style_1-2.css">
		<meta itemprop="name" content="Home and Learn">
		<meta itemprop="description" content="Free C# .NET Course">
		<meta itemprop="image" content="../logo5.jpg">
		<link rel="publisher" href="https://plus.google.com/104045712242599534737" />
		<link rel="author" href="https://plus.google.com/104045712242599534737" />
		<SCRIPT LANGUAGE="JavaScript" type="text/javascript" src="links.js"></SCRIPT>
	</HEAD>


	<BODY>
		<HEADER ID="Banner"><P>&nbsp;</P>
			<H1><SPAN class="Logo_Text">Free</SPAN> <SPAN class="Logo_Text2">C#.NET</SPAN> <SPAN class="Logo_Text">Course</SPAN></H1>
		</HEADER>
<HR />

		<NAV ID="left_menu">
		<P>&nbsp;</P>
<P>&nbsp;</P>
		<H3>Computer Courses</H3>
<ul>
<LI><a href="../index-2.html">Home</a></LI>
<LI><a href="../BC/BeginnersComputing.html">Beginners Computing </a></LI>
<LI><a href="../word2007_2010/Word-2007-2010.html">Word 2007 to 2010</a></LI>
<LI><a href="../MW/MicrosoftWord.html">Word 2000 to 2003</a></LI>
<LI><a href="../excel2007/Excel2007.html">Excel 2007 to 2010</a></LI>
<LI><a href="../ME/MicrosoftExcel.html">Excel to 2003</a></LI>
<LI><a href="../WD/WebDesign.html">Web Design</a></LI>
<LI><a href="../javascript/javascript.html">Javascript</a></LI>
<LI><a href="../NET/vbNet.html">Visual Basic .NET</a></LI>
<LI><a href="../php/php.html">Beginners PHP</a></LI>
<LI><a href="csharp.html">C# .NET</a> <B>&lt; &lt;</B></LI>
<LI><a href="../java/java.html">Java for Beginners</a></LI>
</ul>
		
</NAV> <SECTION ID="Wrapper"> <ARTICLE> <HEADER> 
<H1 ID="Article">Try ... Catch in C# .NET</H1>
</HEADER> <SECTION> 
<p align="center"> 
  <script language="JavaScript">
				writeImageCsharp()
			</script>
</p>
<p>&nbsp;</p>
<p align="center"><a href="csharp_s5p5.html">&lt;&lt; Continues from the previous 
  lesson</a></p>
<p align="left">C# has some inbuilt objects you can use to deal with any potential 
  errors in your code. You tell C# to <b>Try</b> some code, and if can't do anything 
  with it you can <b>Catch</b> the errors. Here's the syntax:</p>
<p align="left" class="indent_small">try<br>
  {</p>
<p align="left" class="indent_small">}<br>
  catch<br>
  {</p>
<p align="left" class="indent_small">}</p>
<p>In the code below, we're trying to load a text file into a RichTextBox called 
  <b>rtb</b>:</p>
<p class="indent_small">try<br>
  {</p>
<p class="indent_big">rtb.LoadFile(&quot;C:/test.txt&quot;);</p>
<p class="indent_small">}<br>
  catch (System.Exception excep)<br>
  {</p>
<p class="indent_big">MessageBox.Show(excep.Message);</p>
<p class="indent_small">}</p>
<p>The code we want to execute goes between the curly brackets of try. We know 
  that files go missing, however, and want to trap this &quot;File not Found&quot; 
  error. We can do that in the catch part. Note what goes between the round brackets 
  after catch:</p>
<p align="center">System.<b>Exception</b> excep</p>
<p><b>Exception</b> is the inbuilt object that handles errors, and this follows 
  the word <b>System</b> (known as a namespace). After <b>System.Exception</b>, 
  you type a space. After the space, you need the name of a variable (<b>excep</b> 
  is just a variable name we made up and, like all variable names, you call it 
  just about anything you like).</p>
<p>The code between the curly brackets of catch is this:</p>
<p align="center">MessageBox.Show(<b>excep.Message</b>);</p>
<p>So we're using a message box to display the error. After the variable name 
  (excep for us), type a dot and you'll see the IntelliSense list appear:</p>
<p align="center"><img src="images/debugging/tryCatch.gif" width="320" height="287" alt="The IntelliSense list"></p>
<p>If you just want to display the inbuilt system message, select <b>Message</b> 
  from the list. When the programme is run, you'll see a message box like this:</p>
<p align="center"><img src="images/debugging/tryCatch2.gif" width="176" height="107" alt="Using the inbuilt System message"></p>
<p>If you know the type of error that will be generated, you can use that instead:</p>
<p class="indent_small">catch (System.IO.FileNotFoundException)<br>
  {</p>
<p class="indent_big">MessageBox.Show(&quot;File not found&quot;);</p>
<p class="indent_small">}</p>
<p>To find out what type of error will be generated, use this:</p>
<p class="indent_small">catch (System.Exception excep)<br>
  {</p>
<p class="indent_big_no_bold">MessageBox.Show( <b>excep.GetType().ToString()</b> 
  );</p>
<p><span class="indent_small">}</span><br>
</p>
<p>The message box will tell you what system error is being generated. You can 
  then use this between the round brackets of catch.</p>
<p>If you want to keep things really simple, though, you can miss out the round 
  brackets after catch. In the code below, we're just creating our own error messages:</p>
<p class="indent_small">try<br>
  {</p>
<p class="indent_big">rtb.LoadFile(&quot;C:/test.txt&quot;);</p>
<p class="indent_small">}<br>
  catch<br>
  {</p>
<p class="indent_big">MessageBox.Show(&quot;An error occurred&quot;);</p>
<p class="indent_small">}</p>
<p>You can add more catch parts, if you want:</p>
<p class="indent_small">try<br>
  {</p>
<p class="indent_big">rtb.LoadFile(&quot;C:/test.txt&quot;);</p>
<p class="indent_small">}<br>
  catch<br>
  {</p>
<p class="indent_big">MessageBox.Show(&quot;An error occurred&quot;);</p>
<p class="indent_small">}<br>
  catch<br>
  {</p>
<p class="indent_big">MessageBox.Show(&quot;Couldn't find the file&quot;);</p>
<p class="indent_small">}<br>
  catch<br>
  {</p>
<p class="indent_big">MessageBox.Show(&quot;Or maybe it was something else!&quot;);</p>
<p class="indent_small">}</p>
<p>The reason you would do so, however, is if you think more than one error may 
  be possible. What if the file could be found but it can't be loaded into a RichTextBox? 
  In which case, have two catch blocks, one for each possibility.</p>
<p>There's also a Finally part you can add on the end:</p>
<p class="indent_small_no_bold">try<br>
  {</p>
<p class="indent_big_no_bold">rtb.LoadFile(&quot;C:/test.txt&quot;);</p>
<p class="indent_small_no_bold">}<br>
  catch (System.Exception excep)<br>
  {</p>
<p class="indent_big_no_bold">MessageBox.Show(excep.Message);</p>
<p class="indent_small">}<br>
  finally <br>
  {</p>
<p class="indent_big">//CLEAN UP CODE HERE</p>
<p class="indent_small">}</p>
<p>You use a <b>Finally</b> block to clean up. (For example, you've opened up 
  a file that needs to be closed.) A Finally block will always get executed, whereas 
  only one catch will.</p>
<p>(NOTE: there is also a <b>throw</b> part to the <b>catch</b> blocks for when 
  you want a more specific error, want to throw the error back to C#, or you just 
  want to raise an error without using try &#133; catch. Try not to worry about 
  throw.)</p>
<p></p>
<p>We won't be using Try &#133; Catch block too much throughout this book, however, 
  because they tend to get in the way of the explanations. But you should try 
  and use them in your code as much as possible, especially if you suspect a particular 
  error may crash your programme.</p>
</SECTION></ARTICLE></SECTION><FOOTER> 
<p align="right"><a href="csharp_s5p5.html">&lt;-- The Locals Window</a> | <a href="csharp_s6p1.html">C# Methods --&gt;</a></p>
		<P><A HREF="csharp.html">Back to the C# NET Contents Page</A>
      		<P>&copy; All course material copyright Home and Learn</P>
    	</FOOTER>

  </BODY>

<!-- Mirrored from www.homeandlearn.co.uk/csharp/csharp_s5p6.html by HTTrack Website Copier/3.x [XR&CO'2005], Thu, 24 Jan 2013 06:55:55 GMT -->
</HTML>

